dora/src/bytecode/generator.rs

```diff
@@ -1001,6 +1008,21 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
         }
     }
 
+    fn emit_intrinsic_array_len(&mut self, arr: &Expr, pos: Position, dest: DataDest) -> Register {
+        if dest.is_effect() {
+            self.visit_expr(arr, dest);
```

Comment:
I don't think this is correct, e.g. `nil.length();` should still throw but AFAIU it doesn't

Reply:
As we talked about a explicit `NilCheck`: Wouldn't it a possibility to use this here? We aren't interested in the result, but only want to check, if the object is nil.